packages/edge-login-ui-rn/src/components/themed/ThemedInputModal.js

```diff
@@ -0,0 +1,143 @@
+// @flow
+
+import * as React from 'react'
+import { View } from 'react-native'
+import { type AirshipBridge } from 'react-native-airship'
+import { cacheStyles } from 'react-native-patina'
+
+import s from '../../common/locales/strings.js'
+import {
+  type Theme,
+  type ThemeProps,
+  withTheme
+} from '../services/ThemeContext.js'
+import { EdgeText } from './EdgeText.js'
+import { EdgeTextField } from './EdgeTextField.js'
+import { PrimaryButton, SecondaryButton } from './ThemedButtons.js'
+import { ThemedModal } from './ThemedModal.js'
+
+type OwnProps = {
+  bridge: AirshipBridge<any>,
+  buttonLabel: string,
+  closeable?: boolean,
+  inputLabel: string,
+  message: string,
+  isInputPassword?: boolean,
+  handleSubmit(string): Promise<any>,
+  title: string
+}
+
+type State = {
+  input: string,
+  error?: string
+}
+
+type Props = OwnProps & ThemeProps
+
+class ThemedInputModalComponent extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props)
+    this.state = {
+      input: ''
+    }
+  }
+
+  async processSubmit() {
+    const { handleSubmit, bridge } = this.props
+    this.setState({ error: undefined })
+    try {
+      const result = await handleSubmit(this.state.input)
+      if (result) {
+        bridge.resolve(this.state.input)
+      }
+    } catch (error) {
+      this.setState({ error: error.message })
+      console.log(error)
+    }
+  }
+
+  handleInputChange = (input: string) => {
+    this.setState({ input })
+  }
+
+  // Needed as PrimaryButton does not accept Promised functions
+  handleSubmit = () => {
+    this.processSubmit()
```

Comment:
Can you move the `catch` block out of `processSubmit`, and put it here as a `.catch(error => { ... })` chain? Anytime a sync method calls an async method, I expect to see a `.catch` block, or there might be an unhandled exception bug in our code.